### PR TITLE
Unittest for MatrixBridge

### PR DESCRIPTION
**Pull request by naveennvrgup**
fixes: JENKINS-59603

Had a little trouble stubbing the chained function calls. Got a way around it. Happy to take remarks :smiley_cat:

### CODE REVIEW

**Comment by uhafner** [comment]
Can you please have a look at the failing checks?

File: plugin/src/test/java/io/jenkins/plugins/analysis/core/steps/MatrixBridgeTest.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+ * Tests the class {@link MatrixBridge}.
+ *
+ * @author Naveen Sundar
+ */
+public class MatrixBridgeTest {
+    // to be tested
+    private MatrixBridge matrixBridge;
+
+    // Dependencies
+    private MatrixBuild build;
```
Comment (uhafner) [inline plugin/src/test/java/io/jenkins/plugins/analysis/core/steps/MatrixBridgeTest.java:26]: Fields are not allowed in tests. Can you please refactor and inline in a method that is called by both tests? You may need to inline `TaskListener.NULL` and launcher as well.

File: plugin/src/test/java/io/jenkins/plugins/analysis/core/steps/MatrixBridgeTest.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        when(matrixProject.getPublishersList()).thenReturn(describableList);
+    }
+
+    @Test
+    public void constructMatrixAggregatorWithoutRecorder() {
+        when(describableList.get(IssuesRecorder.class)).thenReturn(null);
+
+        MatrixAggregator aggregator = matrixBridge.createAggregator(build, launcher, listener);
+
+        assertNull(aggregator);
```
Comment (uhafner) [inline plugin/src/test/java/io/jenkins/plugins/analysis/core/steps/MatrixBridgeTest.java:57]: We are using AssertJ in the whole project.

**Review by uhafner — CHANGES_REQUESTED** [review]

**Comment by naveennvrgup** [comment]
> Can you please have a look at the failing checks?

Sure I will look into the issue immediately

File: plugin/src/test/java/io/jenkins/plugins/analysis/core/steps/MatrixBridgeTest.java (hunk 2 above)
Comment (naveennvrgup) [reply plugin/src/test/java/io/jenkins/plugins/analysis/core/steps/MatrixBridgeTest.java]: got it :+1: . AssertJ it is

File: plugin/src/test/java/io/jenkins/plugins/analysis/core/steps/MatrixBridgeTest.java (hunk 1 above)
Comment (naveennvrgup) [reply plugin/src/test/java/io/jenkins/plugins/analysis/core/steps/MatrixBridgeTest.java]: ok

**Comment by naveennvrgup** [comment]
The build is failing now due to loss of connection with EC2 windows machine. I think it has nothing to do with the unittest which I have written. Please check. Thanks

image

**Review by uhafner — APPROVED** [review]

**Comment by uhafner** [comment]
Thanks!

**Pull request merged**